springfox-spring-web/src/main/java/springfox/documentation/spring/web/plugins/JavaDocBuilderPlugin.java

```diff
@@ -0,0 +1,158 @@
+/*
+ *
+ *  Copyright 2015 the original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ */
+package springfox.documentation.spring.web.plugins;
+
+/**
+ * Plugin to generate the @ApiParam and @ApiOperation values.
+ */
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+import springfox.documentation.service.ResolvedMethodParameter;
+import springfox.documentation.spi.DocumentationType;
+import springfox.documentation.spi.service.OperationBuilderPlugin;
+import springfox.documentation.spi.service.ParameterBuilderPlugin;
+import springfox.documentation.spi.service.contexts.OperationContext;
+import springfox.documentation.spi.service.contexts.ParameterContext;
+
+/**
+ * Swagger JavaDoc properties builder plugin.
+ */
+@Component
+@Order(Ordered.LOWEST_PRECEDENCE)
+public class JavaDocBuilderPlugin implements OperationBuilderPlugin, ParameterBuilderPlugin {
+
+    private static final String PERIOD = ".";
+    private static final String API_PARAM = "io.swagger.annotations.ApiParam";
+    private static final String REQUEST_PARAM = "org.springframework.web.bind.annotation.RequestParam";
+    private static final String PATH_VARIABLE = "org.springframework.web.bind.annotation.PathVariable";
+
+    @Autowired
+    private JavaDocPropertiesReader propertiesReader;
+
+    @Override
+    public boolean supports(DocumentationType delimiter) {
+        return true;
+    }
+
+    @Override
+    public void apply(OperationContext context) {
+
+        String notes = context.requestMappingPattern() + PERIOD +
+                context.httpMethod().toString() + ".notes";
+        if (StringUtils.hasText(notes) && StringUtils.hasText(propertiesReader.getProperty(notes))) {
+            context.operationBuilder().notes(propertiesReader.getProperty(notes));
+        }
+    }
+
+    @Override
+    public void apply(ParameterContext context) {
```

Comment:
Hmm this reveals an interesting gap. Really in this case you should have an API provided by springfox like `ParameterContext#updateProperties(Environment env)` So the context can load the properties it cares about like description etc. Remember it already knows its name/isrequired etc from other plugins reading the spring annotations :)